api/v1alpha1/stardogrole_types.go

```diff
@@ -25,19 +26,44 @@ import (
 
 // StardogRoleSpec defines the desired state of StardogRole
 type StardogRoleSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// RoleName describes the name of a Role.
+	RoleName string `json:"roleName,omitempty"`
+	// StardogInstance describes in which Stardog Instance the user must be maintained
+	StardogInstance string `json:"stardogInstance,omitempty"`
+	// Permissions lists the permissions assigned to a role
+	Permissions []StardogPermissionSpec `json:"permissions,omitempty"`
+}
 
-	// Foo is an example field of StardogRole. Edit StardogRole_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+// StardogPermissionSpec defines a Stardog permission assigned to a Role
+type StardogPermissionSpec struct {
+	// Action describes the action a specific permission is assigned to
+	// Types of Actions: ALL, CREATE, DELETE, READ, WRITE, GRANT, REVOKE, EXECUTE
+	Action string `json:"action,omitempty"`
+	// ResourceType describes the type of resource a specific permission is assigned to
+	// Types of Resources: DB, USER, ROLE, ADMIN, METADATA, NAMEDGRAPH, VIRTUALGRAPH, ICVCONTRAINTS
+	ResourceType string `json:"resourceType,omitempty"`
+	// Resources is a list of permission objects that get each targeted by the action and resource type properties
+	Resources []string `json:"resources,omitempty"`
 }
 
 // StardogRoleStatus defines the observed state of StardogRole
 type StardogRoleStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Conditions contain the states of the StardogRole. A StardogRole is considered Ready when the role has been
+	// persisted to Stardog DB.
+	Conditions []StardogRoleCondition `json:"conditions,omitempty" patchStrategy:"merge"`
+}
+
+// StardogRoleCondition describes a status condition of a StardogRole
+type StardogRoleCondition struct {
```

Comment:
We can have a single Condition struct for all Stardog conditions, no need to have specific structs for instances, roles and users. In K8s v1.19, the Condition struct is part of the meta v1 API, so then we don't even need this anymore.
```suggestion
type StardogCondition struct {
```